static/app/components/dropdownControl.tsx

```diff
@@ -97,6 +97,7 @@ class DropdownControl extends React.Component<Props> {
             priority={priority}
             {...getActorProps(buttonProps)}
             isOpen={isOpen}
+            data-test-id="dropdown-control-button"
```

Comment:
Why can't we use the more semantic getByRole type stuff instead of using test-ids?

Reply:
`getByRole` is possibly slower because I _think _  it might use visual styles to check for hidden, I was just checking if test-ids were that much faster, and didn't want other sources of visual style computing (for hidden check) when I was profiling.

Reply:
You're correct it is slower. When should we be using what 🤔

Reply:
Just use test-ids. It's easy to find them later programmatically to replace vs. all other options since it's data unique to tests.

Reply:
I wonder if `getByRole` performs better on smaller DOM trees? It would be good to have consistent direction to developers on when to use each kind of query.